src/View/Form/EntityContext.php

```diff
@@ -201,16 +201,26 @@ public function isCreate()
      * Traverses the entity data and finds the value for $path.
      *
      * @param string $field The dot separated path to the value.
+     * @param array $options Options:
+     *   - `default`: Default value to return if no value found in request
+     *     data or entity.
+     *   - `schemaDefault`: Boolen indicating whether default value from table
+     *     schema should be used if it's not explicitly provided.
      * @return mixed The value of the field or null on a miss.
      */
-    public function val($field)
+    public function val($field, $options = [])
```

Comment:
Typehints have caused us quite a bit of grief, so for methods like this which are generally not going to be directly used by users i would prefer not adding typhints.